fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java

```diff
@@ -37,18 +39,22 @@ public ProcResult fetchResult() {
             return result;
         }
 
-        // get info
-        List<List<Comparable>> dbInfos = new ArrayList<List<Comparable>>();
+        List<List<Comparable>> dbInfos = new ArrayList<>();
+
+
+        if (!Config.disable_check_db_status) {
+            // if a large number of db under the catalog, this operation is very slow
+            dbNames = dbNames.stream().filter(dbName -> {
```

Comment:
Why the operation is slow? 8000 databases is not very large. The iteration is so slow?